src/context/cart/cartContext.js

```diff
@@ -0,0 +1,99 @@
+import React, { createContext, useContext, useReducer } from "react";
+import { useEffect } from "react";
+import {
+  getCartApi,
+  addToCartApi,
+  updateQuantityApi,
+  removeFromCartApi,
+} from "../../services/cart_services";
+import { price } from "../../shared/utils/function";
+import { reducer } from "./cartReducer";
+
+const CartContext = createContext();
+
+export const CartContextProvider = ({ children }) => {
+  useEffect(() => {
+    if (status !== "IDLE") return;
+    (async function () {
+      dispatch({ type: "UPDATE_STATUS", payload: { status: "PENDING" } });
+      const { data, success, message } = await getCartApi();
+
+      if (success) {
+        dispatch({ type: "UPDATE_CART", payload: { cart: data.cart } });
+        dispatch({ type: "UPDATE_STATUS", payload: { status: "SUCCESS" } });
+      }
+      dispatch({ type: "UPDATE_STATUS", payload: { status: "FAILED" } });
```

Comment:
fix: this dispatch should be in else block, otherwise it would always be triggered

Reply:
done !!